### PR TITLE
Interfaces sub-command completed w/ add/list/remove/update support.

### LINKED ISSUE #68: Finish implementing support for Interfaces in CLI
This includes:

- update action
- remove action
- add remaining search fields for list action
- list interface addresses (sub-command)
- list interface assignments (sub-command)
- list all interfaces for a device (from devices `nsot devices -H foo-bar1 interfaces`

### LINKED ISSUE #75: Unit test *all* actions for all resources
Embarrassing but due to lack of resources, that's where we are! That's add/update/list/remove for everything.

### PR DESCRIPTION
**Pull request by jathanism**
fixes #68
fixes #75

NOTE: Interfaces cannot yet be added/updated/deleted by natural key
because it requires a backend REST API change. We may not get to this
before v1.0.

- Updated to support NSoT 0.17.2.
- For Networks, Devices, and Interfaces the following attribute flags
  have been renamed across all sub-commands for better consistency of
  argument namespace across all commands:
  - `-R` for `--replace-attribute` (was `-r`)
  - `-D` for `--delete-attributes` (was `-d`)
  - `-M` for `--multi` (was `-m`)
- Allowed natural_key lookup for interfaces to support either device_id
  or (device_id, name)
- Removed requests_mock from dev requirements
- Updated requirements: `requests==2.9.1`
- Tons of documentation cleanup to make using NSoT a breeze!
- A few minor PEP8 cleanups.

### CODE REVIEW

**Comment by coxley** [comment]
Only partially read through the code, but the documentation was engaging I thought. Maybe include some occurrences of `--grep`?

I really like the derrived `networks` and `assignments` list subcommands. In addition to what's there I would like to see more human names vs. referencing IDs but I understand it might need backend adjustments.

**Comment by jathanism** [comment]
@coxley Thanks for the feeback. The `-g/--grep` flag hasn't been documented anywhere yet. 

I want to move the output modifiers to their own section... `-N/--natural-key`, -`g/--grep`, `-d/--delimited` since they are common to all resources. I'll get to that tomorrow! zzz

File: pynsot/commands/cmd_interfaces.py [hunk 1]
```diff
@@ -67,53 +71,107 @@ def cli(ctx):
...
-    '-b',
-    '--bulk-add',
-    metavar='FILENAME',
-    help='Bulk add Networks from the specified colon-delimited file.',
-    type=click.File('rb'),
-    callback=callbacks.process_bulk_add,
+    '-c',
+    '--addresses',
+    metavar='ADDRS',
+    help='An IPv4/IPv6 addres in CIDR format.',
```
Comment (dmar42) [inline pynsot/commands/cmd_interfaces.py]: `address` not `addres`

File: pynsot/commands/cmd_interfaces.py [hunk 2]
```diff
@@ -67,53 +71,107 @@ def cli(ctx):
...
-    You must provide a Device ID using the -d/--device-id option.
+    You must provide a Device ID using the -D/--device option.
 
     When adding a new Interface, you must provide a value for the -n/--name
     option.
 
     If you wish to add attributes, you may specify the -a/--attributes
     option once for each key/value pair.
+
+    If you wish to assign addresses, you may specify the -A/--addresses
```
Comment (dmar42) [inline pynsot/commands/cmd_interfaces.py]: This is `-c` now, per line 79 above.

File: pynsot/commands/cmd_interfaces.py [hunk 3]
```diff
@@ -324,28 +416,76 @@ def remove(ctx, id, site_id):
...
-    help='A key/value pair attached to this Network (format: key=value).',
+    help='A key/value pair attached to this Interface (format: key=value).',
     multiple=True,
     callback=callbacks.transform_attributes,
 )
 @click.option(
+    '-c',
+    '--addresses',
+    metavar='ADDRS',
+    help='An IPv4/IPv6 addres in CIDR format.',
```
Comment (dmar42) [inline pynsot/commands/cmd_interfaces.py]: Fix `address` here as well

File: pynsot/commands/cmd_interfaces.py (hunk 1 above)
Comment (jathanism) [inline pynsot/commands/cmd_interfaces.py]: syn-ack

File: pynsot/commands/cmd_interfaces.py (hunk 2 above)
Comment (jathanism) [inline pynsot/commands/cmd_interfaces.py]: syn-ack

File: pynsot/commands/cmd_interfaces.py (hunk 3 above)
Comment (jathanism) [inline pynsot/commands/cmd_interfaces.py]: syn-ack. (and hey at least I'm consistent)

**Comment by coxley** [comment]
This is really great!

**Pull request merged**